src/fileformat/types/import_table/elf_import_table.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file include/retdec/fileformat/types/import_table/elf_import_table.h
+ * @brief Class for ELF import table.
+ * @copyright (c) 2021 Avast Software, licensed under the MIT license
+ */
+
+#include "retdec/fileformat/utils/crypto.h"
+#include "retdec/utils/string.h"
+#include "retdec/fileformat/utils/tlsh/tlsh.h"
+#include "retdec/fileformat/types/import_table/elf_import_table.h"
+#include <algorithm>
+#include <regex>
+
+using namespace retdec::utils;
+
+namespace retdec {
+namespace fileformat {
+
+static const std::set<std::string> exclusion_set = {
+	"__libc_start_main", // main function
+	"main", // main function
+	"abort", // ARM default
+	"cachectl", // MIPS default
+	"cacheflush", // MIPS default
+	"puts", // Compiler optimization (function replacement)
+	"atol", // Compiler optimization (function replacement)
+	"malloc_trim" // GNU extensions
+};
+
+bool isSymbolExcluded(std::string symbol)
+{
+	if (symbol.empty()) {
+		return true;
+	}
+	/* ignore:
+		symbols starting with . or _
+		x86-64 specific functions
+		string functions (str.* and mem.*), gcc changes them depending on architecture
+		symbols starting with . or _
+	*/
+	std::regex exclusion_regex("(^[_\.].*$)|(^.*64$)|(^str.*$)|(^mem.*$)");
+
+	if (std::regex_match(symbol, exclusion_regex)) {
+		return true;
+	}
+
+	if (exclusion_set.count(symbol)) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * Compute import telfhash
+ */
+void ElfImportTable::computeHashes()
+{
+	std::vector<std::string> imported_symbols;
+
+	for (const auto& import : imports) {
+		auto funcName = toLower(import->getName());
+
+		// filter the symbols just as telfhash does for the same result
+		if (isSymbolExcluded(funcName)) {
+			continue;
+		}
+
+		imported_symbols.push_back(funcName);
+	}
+
+	std::sort(imported_symbols.begin(), imported_symbols.end());
```

Comment:
If you are sorting, why not use `std::set`? Are duplicates possible/desirable? If yes, and we want to keep them, `std::vector` is ok. If we don't want them, `std::set` would solve it for us.
Does telfhash  allow duplicates in this container?

Reply:
Also a thing, is the sorting algorithm behaving the same as in the telfhash implementation? I think it is possible to somehow control the string sorting in C++? There might be differences. And even it there are not, a comment on this would be helpful so that no one screws it up later.

Reply:
Original implementation seems to use a python list and not check for duplicates, so I think it allows duplicates. Python sort and C++ one on strings should sort lexicographically. But I'll be more sure after further testing.